autoclass_migration/migrate_bucket.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+
+set -o pipefail
+set -ex
+
+if [[ -z "$GCP_PROJECT" ]]; then
+    echo "No GCP project specified"
+    exit 1
+fi
+
+if [[ -z "$BUCKET" ]]; then
+    echo "No bucket specified"
+    exit 1
+fi
+
+if [[ -z "$BILLING_PROJECT" ]]; then
+    echo "No billing project specified"
+    exit 1
+fi
+
+if [[ -z "$SLACK_WEBHOOK" ]]; then
+    echo "No Slack webhook specified"
+    exit 1
+fi
+
+post_to_slack() {
+    curl --fail --silent --show-error  -X POST \
+        -H 'Content-type: application/json' \
+        -d '{"text":"'"$1"'"}' \
+        "$SLACK_WEBHOOK"
+}
+
+report_exit_status() {
+    rv=$?
+
+    if [[ $rv = 0 ]]; then
+        post_to_slack ":white_check_mark: migration for $BUCKET finished successfully"
+    else
+        post_to_slack ":x: migration for $BUCKET failed"
+    fi
+
+    exit $rv
+}
+
+trap "report_exit_status" EXIT
+
+post_to_slack "Starting migration for $BUCKET"
+
+# First check that Autoclass hasn't already been enabled.
+if gsutil -u "$BILLING_PROJECT" autoclass get "gs://$BUCKET" | grep True; then
+    post_to_slack "Autoclass is already enabled for $BUCKET"
+    exit 0
+fi
+
+# Store the IAM permissions.
+gsutil -u "$BILLING_PROJECT" iam get "gs://$BUCKET" > /tmp/iam.json
+
+# Store the object versioning setting.
+OBJECT_VERSIONING=$(gsutil -u "$BILLING_PROJECT" versioning get "gs://$BUCKET" | cut -f 2 -d ' ')
+
+# Store the lifecycle configuration.
+gsutil -u "$BILLING_PROJECT" lifecycle get "gs://$BUCKET" > /tmp/lifecycle_config.json
+
+# Store the Requester Pays setting.
+REQUESTER_PAYS=$(gsutil -u "$BILLING_PROJECT" requesterpays get gs://cpg-fewgenomes-test | cut -f 2 -d ' ')
+
+# Remove all IAM permissions to prevent modifications while we perform the temporary copy.
+gsutil -u "$BILLING_PROJECT" iam set -e '' <(echo "{}") "gs://$BUCKET"
+
+# Determine total size.
+BUCKET_SIZE=$(gsutil -u "$BILLING_PROJECT" du -s "gs://$BUCKET" | cut -f 1 -d ' ')
+post_to_slack "Bucket size for $BUCKET: $BUCKET_SIZE B"
```

Comment:
Could we report this in GB? I'm getting confused my large numbers.
```suggestion
post_to_slack "Bucket size for $BUCKET: $((BUCKET_SIZE / 1073741824)) GB"
```

Reply:
Found something even better! :)